certexchange/client.go

```diff
@@ -56,11 +59,23 @@ func (c *Client) Request(ctx context.Context, p peer.ID, req *Request) (_rh *Res
 		}
 	}()
 
+	requestStart := time.Now()
+	var dialSuceeded bool
+	defer func() {
+		d := float64(time.Since(requestStart)) / float64(time.Second)
+		metrics.requestLatency.Record(ctx, d, metric.WithAttributes(
+			mhelper.Status(ctx, _err),
+			mhelper.AttrDialSucceeded.Bool(dialSuceeded),
```

Comment:
Separate attribute because I want to track whether or not we failed in the dial and _why_ we failed.